### ISSUE TITLE
Optional disable link preview in all your new messages

### ISSUE THREAD

**Issue by hyperion-cs** [telegramdesktop/tdesktop#28725]
### Is your feature request related to a problem?

I don't like the link preview widget in messages (and I believe I'm not the only one).
I mean, when you write a message and insert a link, this widget appears:
<img width="701" alt="image" src="https://github.com/user-attachments/assets/d3c4f2eb-de4c-40b1-b2d6-fdcbb963771f">
Which when you send a message turns into:
<img width="491" alt="image" src="https://github.com/user-attachments/assets/f57e74ac-1d6e-4a32-adf8-440ce90fd62f">

This is not only not to everyone's liking (e.g., it takes up a lot of space in the chat room), but also inappropriate when you have several links in a message (including not only "raw" links, but also links in the form of clickable text via _Formatting -> Create Link_).
One message is limited to ONE link preview widget, but at the same time, one message can have SEVERAL links (sometimes dozens of them). Showing link preview for any one of them (chosen by an unknown algorithm) is a **bad** solution, I would prefer that they (widgets) were not there at all. However, there may be different opinions on this, so I want it to be optional.

At the moment I have to edit a message after sending it by MANUALLY removing the link preview (because even if you remove it when you write a message, it may reappear after you send it). And it's like that every time ;(

### Describe the solution you'd like

Add an option (flag) in _Settings -> Chat settings -> Messages_ that would disable link preview widget generation when writing new messages. Older messages that already have this widget should still display it, of course.

### Describe alternatives you've considered

Every time you send a message with links, edit it by MANUALLY removing the link preview widget from it.

### Additional context

This shouldn't be too hard, since it's just adding an option (flag) on the client and then a `if` condition on that flag that checks if the link preview widget generation function should be run (while writing/sending a message).

**Comment by alessandro-saglimbeni** [comment]
repost, this is a pretty important security and privacy enhancement

**Comment by Weltolk** [comment]
we need this

**Comment by July250229** [comment]
I agree. Link preview means we can't paste any one-time url into Telegram

**Comment by i8degrees** [comment]
I humbly agree likewise. Please, offer a toggle to turn this "feature" off.

**Comment by NicolaP8** [comment]
This is the main reason for me to not use Telegram with all my clients and friends, only some collegues.
It's a BIG security and privacy hole.

**Comment by i8degrees** [comment]
I should not have to even consider the possibility of DNS block for preventing this very concerning security hole. Last I checked, not even that is a worth while effort due to how the "feature" works at a technical level.

So, it is positively not an accidental but exactly as intended "feature". It does not have to be this way.

Please, for the love of all things Godly and otherwise, consider a legit opt-out from this anti-feature. I pray somebody with the ability to do this is listening or will at some point soon be listening to us cry out for this. 

I would happily implement the feature for the app 100% and have as close to zero cost as possible to the operators of the Telegram service if only I thought that it would be accepted upstream.

Cheers

**Comment by vyakovlev** [comment]
Support this feature. It's very inconvenient when you send links to local services, that can not be resolved by Telegram.

**Comment by aho-amiblu** [comment]
+1 It is really annoying that this cannot be turned off.

**Comment by pidrilaebanaja** [comment]
I'll just bump this
